tutorials/20_score_based_methods_new_features.ipynb

```diff
@@ -0,0 +1,710 @@
+      "Drawing 100 posterior samples: 100%|██████████| 499/499 [00:03<00:00, 148.17it/s]\n",
+      "Drawing 10000 posterior samples: 10099it [00:20, 486.58it/s]\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Build the posterior using the trained score estimate, and sample with the sde option\n",
+    "posterior = inference.build_posterior(score_estimator, sample_with=\"sde\")\n",
+    "samples = posterior.sample((10000,), x=x_o, predictor=\"euler_maruyama\", corrector=\"langevin\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The argument `sample_with` allows to choose which kind of solver we want to use to reverse the diffusion equation :\n",
+    "\n",
+    "- `\"ode\"` : builds the probability flow ODE using the zuko library. We can sample from the flow afterwards by calling `.sample()`\n",
+    "\n",
+    "- `\"sde\"` : solves the reverse SDE by alternating **prediction** and **correction** steps  \n",
+    "\n",
+    "Both options contain **rejection sampling** steps to ensure prior coverage.  \n",
+    "Note that in the `\"sde\"` option, the only available predictor is `\"euler_maruyama\"` and the available correctors are `\"langevin\"` (for **Unadjusted Langevin Dynamics**) and `\"gibbs\"`.  \n",
+    "\n",
+    "The argument `x_o` in the `.sample()` method indicates which observation the posterior should be evaluated at before sampling and replaces the `.set_default()`method."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### MAP estimation\n",
+    "It is possible to get a Maximum A Posterior (MAP) estimate in the case of score-based methods. Note that in this case, you should **first** evaluate the posterior at the desired observation, using the `set_default_x` method !  \n",
+    "The optimization procedure can be interrupted by the user : the best estimate will still be saved and can be accessed with `.map()`.  \n",
+    "\n",
+    "The default values used by this function (e.g. number of optimization steps, number of starting positions, learning rate) might require hand-tuning from the user for the problem at hand."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 83,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "Drawing 1000 posterior samples: 100%|██████████| 499/499 [00:01<00:00, 451.12it/s]\n",
+      "Drawing 1000 posterior samples: 100%|██████████| 1000/1000 [00:01<00:00, 895.55it/s]\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "tensor([[-0.1480,  0.0664]])\n"
+     ]
+    }
+   ],
+   "source": [
+    "# MAP estimation\n",
+    "posterior.set_default_x(x_o)\n",
+    "map_est = posterior.map(num_iter=5)\n",
```

Comment:
please add a comment what `num_iter=5` means here and why it is fine to set it to 5